### PR TITLE
Fix Issue #85

### LINKED ISSUE #85: pointerInput not working in MarkdownText
```
 MarkdownText(
            markdown = message,
            modifier = Modifier.pointerInput(Unit) {
                detectTapGestures(
                    onLongPress = {
                        Log.d("TAG", "long pressed")
                    }
                )
            }
        )
```

This doesn't work, if I use normal text the above code works perfectly.

### PR DESCRIPTION
**Pull request by HXavier21**
Fix #85 by adding a CustomTextView, which will continue passing the event to other components when there is no link clicked.
Thus the below code can work as well:
```kotlin
MarkdownText(
    markdown = """
           ---
            __Advertisement :)__
            
            - __[nodeca](https://nodeca.github.io/pica/demo/)__ - high quality and fast image
              resize in browser.
            - __[babelfish](https://github.com/nodeca/babelfish/)__ - developer friendly
              i18n with plurals support and easy syntax.
            
            You will like those projects!
            
            ---
        """.trimIndent(),
    modifier = Modifier.pointerInput(Unit) {
        detectTapGestures(onLongPress = {
            Toast
                .makeText(this@MainActivity, "On long press", Toast.LENGTH_SHORT)
                .show()
        })
    }
)
```

### CODE REVIEW

**Review by jeziellago — CHANGES_REQUESTED** [review]
Hi @HXavier21, thanks for contributing!
I appreciate your contribution, however there are some opportunities to improve. 
1. All code should be written in Kotlin
2. Every change should provide backward compatibility, it means code being removed/deleted usually force the library users to change their code when update
I understand your use case, but we should think in other use cases that might break when a new change is introduced.

File: markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package dev.jeziellago.compose.markdowntext
+
+import android.content.Context
+import android.text.Spannable
+import android.text.style.ClickableSpan
+import android.view.MotionEvent
+import androidx.appcompat.widget.AppCompatTextView
+
+class CustomTextView(context: Context?) : AppCompatTextView(context!!) {
```
Comment (jeziellago) [inline markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt:9]: The `Context` here should be required.
```suggestion
class CustomTextView(context: Context) : AppCompatTextView(context) {
```

File: markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        val off = layout.getOffsetForHorizontal(line, x.toFloat())
+
+        val spannable = text as Spannable
+        return spannable.getSpans(off, off, ClickableSpan::class.java)
+    }
+
+    override fun performClick(): Boolean {
+        return super.performClick()
+    }
+}
```
Comment (jeziellago) [inline markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt:46]:
```suggestion
}

```

File: markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt (hunk 1 above)
Comment (HXavier21) [reply markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt:9]: didn't notice that😢

File: markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt (hunk 2 above)
Comment (HXavier21) [reply markdowntext/src/main/java/dev/jeziellago/compose/markdowntext/CustomTextView.kt:46]: cannot understand but ok

**Review by jeziellago — APPROVED** [review]

**Comment by HXavier21** [comment]
Find a fatal error, used a full-width punctuation in MainActivity while edit code in Github. Sorry for that.😭

**Review by jeziellago — APPROVED** [review]

**Comment by AleksandarIlic** [comment]
@HXavier21 @jeziellago This changes broke text selection. It's impossible to select text now.

**Comment by HXavier21** [comment]
> @HXavier21 @jeziellago This changes broke text selection. It's impossible to select text now.

I'll check it right away😢

**Comment by HXavier21** [comment]
> @HXavier21 @jeziellago This changes broke text selection. It's impossible to select text now.

It seems that selecting text also needs to trigger a long press event, causing a conflict.
A workable but not good modification is：
```kotlin
when (isTextSelectable) {
    true -> TextView(factoryContext)
    false -> CustomTextView(factoryContext)
}.apply {
    ...
}
```
Another possible approach is to add control of `isTextSelectable` in `CustomTextView` and adapt the `MovementMethod` of selecting text in `OnTouchEvent`.
Are there any other methods?

**Comment by HXavier21** [comment]
Since enabling text selection means disabling the long press gesture, it has been proven in practice that the bug can be simply fixed by the following changes:

**Comment by jeziellago** [comment]
@HXavier21 could open a PR fixing it?

**Comment by HXavier21** [comment]
> @HXavier21 could open a PR fixing it?
> In case you can't, I'll revert this change until we have a new and well-tested solution that doesn't break the current impl.

I'll open a PR right away, and I'd like to know if there are any bugs similar to this one🤔

**Comment by HXavier21** [comment]
> @HXavier21 could open a PR fixing it? In case you can't, I'll revert this change until we have a new and well-tested solution that doesn't break the current impl.

See #108

**Pull request merged**